clientlibs/js/src/ApiService/ApiService.ts

```diff
@@ -0,0 +1,318 @@
+import { UpGradeClientEnums, UpGradeClientInterfaces, UpGradeClientRequests } from '../types';
+import { DefaultHttpClient } from '../DefaultHttpClient/DefaultHttpClient';
+import {
+  CaliperEnvelope,
+  IExperimentAssignmentv5,
+  IFeatureFlag,
+  IFlagVariation,
+  IGroupMetric,
+  ILogInput,
+  ISingleMetric,
+} from 'upgrade_types';
+import { DataService } from 'DataService/DataService';
+import { IApiServiceRequestParams, IEndpoints } from './ApiService.types';
+import { IMarkDecisionPointParams } from 'UpGradeClient/UpGradeClient.types';
+import * as uuid from 'uuid';
```

Comment:
As you are only using v4 you can change the import statement to
import { v4 as uuidv4 } from 'uuid';